### PR TITLE
[Tune] Update PBT Transformers Example

### PR DESCRIPTION
**Pull request by amogkam**
## Why are these changes needed?
* Updates PBT Transformers example to use wandb_mixin
* Adds link to Colab notebook on docs
* Small change to wandb integration to check if user is already logged in

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/integration/wandb.py [hunk 1]
```diff
@@ -95,18 +95,26 @@ def _set_api_key(wandb_config):
...
+    try:
+        # Check if user is already logged into wandb.
+        wandb.ensure_configured()
+        if wandb.api.api_key:
+            logger.info("Already logged into W&B.")
+            return
+    except AttributeError:
+        if api_key_file:
+            if api_key:
+                raise ValueError("Both WandB `api_key_file` and `api_key` set.")
```
Comment (richardliaw) [inline python/ray/tune/integration/wandb.py:107]:
```suggestion
                raise ValueError("Both WandB `api_key_file` and `api_key` set.") from NOne
```

Comment (richardliaw) [inline python/ray/tune/integration/wandb.py:107]: shouldn't this happen before the try-catch?

File: python/ray/tune/integration/wandb.py [hunk 2]
```diff
@@ -95,18 +95,26 @@ def _set_api_key(wandb_config):
...
-            "No WandB API key found. Either set the {} environment "
-            "variable or pass `api_key` or `api_key_file` in the config".
-            format(WANDB_ENV_VAR))
+            os.environ[WANDB_ENV_VAR] = api_key
+        elif not os.environ.get(WANDB_ENV_VAR):
+            raise ValueError(
+                "No WandB API key found. Either set the {} environment "
+                "variable, pass `api_key` or `api_key_file` in the config, "
+                "or run `wandb login` from the command line".
+                format(WANDB_ENV_VAR))
```
Comment (richardliaw) [inline python/ray/tune/integration/wandb.py:117]: can we move this to outside of the exception catching block?

File: python/ray/tune/integration/wandb.py (hunk 1 above)
Comment (krfricke) [reply python/ray/tune/integration/wandb.py:107]: I think we will have to discuss the login priorities. Right now, wandb login takes priority over the other methods, however, I think if someone explicitly passes an `api_key_file` or even `api_key` in the config, he would like to log to that account.

IMO, the priority should be like this:

1) Config `api_key`/`api_key_file` (only one allowed)
2) Environment variable
3) wandb login

File: python/ray/tune/integration/wandb.py (hunk 2 above)
Comment (amogkam) [reply python/ray/tune/integration/wandb.py]: done

File: python/ray/tune/integration/wandb.py (hunk 1 above)
Comment (amogkam) [reply python/ray/tune/integration/wandb.py]: Makes sense. I updated it to reflect that logic.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
ping when tests ok?

**Review by krfricke — APPROVED** [review]
LGTM

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py [hunk 3]
```diff
@@ -62,6 +55,8 @@ def recover_checkpoint(tune_checkpoint_dir, model_name=None):
 
 
 # __train_begin__
+# Uncomment this line to use W&B!
+#@wandb_mixin
```
Comment (krfricke) [inline python/ray/tune/examples/pbt_transformers/pbt_transformers.py:59]:
```suggestion
# @wandb_mixin
```

File: python/ray/tune/examples/pbt_transformers/pbt_transformers.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import ray
 from ray.tune import CLIReporter
+from ray.tune.integration.wandb import wandb_mixin
```
Comment (krfricke) [inline python/ray/tune/examples/pbt_transformers/pbt_transformers.py:5]:
```suggestion
from ray.tune.integration.wandb import wandb_mixin  # noqa: F401
```

**Review by richardliaw — APPROVED** [review]

**Pull request merged**